ZenJourney/app/src/main/java/ca/uwaterloo/cs/MainActivity.kt

```diff
@@ -89,9 +91,16 @@ fun loadLocalUserSettings(context: Context, appState: AppState) {
 
 @Composable
 fun MainContent(context: Context) {
-    val appState = remember { AppState() }
+    val appState = remember { AppState(context) }
 
-    loadLocalUserSettings(context, appState)
+    runBlocking {
+        val jwt = appState.dataStore.getJwt()
+        if (jwt.isNotEmpty() && !JWT(jwt).isExpired(5)) {
+            appState.pageState.value = PageStates.HOME
+        }
+    }
+
+    LoadLocalUserSettings(context, appState)
```

Comment:
should we load user settings first or doesn't matter?

Reply:
might matter in the future so i'll move it